PR 910 introduces regression whereby sorting collection throws exceptions at JVM exit

@magicDGS It looks to me like your PR #910 changes the behavior of temporary file creation and cleanup in a way that breaks lots of downstream usage.

The problem appears to be in `DeletePathThread` where it attempts to delete a Path and rethrows any exceptions that are encountered.  The problem is that one of those exceptions that can be thrown is a `NoSuchFileException` because the file has already been cleaned up.

Running SortSam using the latest version of Picard I get:

```
$  java -Xmx4g -jar picard.jar SortSam I=in.bam O=out.bam SO=queryname
...
INFO	2017-11-16 07:50:23	SortSam	Finished reading inputs, merging and writing to output now.
[Thu Nov 16 07:50:27 EST 2017] picard.sam.SortSam done. Elapsed time: 0.18 minutes.
Runtime.totalMemory()=1586495488
Exception in thread "Thread-2" Exception in thread "Thread-1" htsjdk.samtools.util.RuntimeIOException: java.nio.file.NoSuchFileException: /var/folders/mz/34h8j89n1jj2mg6frd0lmqxh0000gn/T/tfenne/sortingcollection.3170510762910173658.tmp
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:374)
Caused by: java.nio.file.NoSuchFileException: /var/folders/mz/34h8j89n1jj2mg6frd0lmqxh0000gn/T/tfenne/sortingcollection.3170510762910173658.tmp
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:372)
htsjdk.samtools.util.RuntimeIOException: java.nio.file.NoSuchFileException: /var/folders/mz/34h8j89n1jj2mg6frd0lmqxh0000gn/T/tfenne/sortingcollection.5241745018661624118.tmp
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:374)
Caused by: java.nio.file.NoSuchFileException: /var/folders/mz/34h8j89n1jj2mg6frd0lmqxh0000gn/T/tfenne/sortingcollection.5241745018661624118.tmp
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:372)
```

This is happening, I believe, because `SortingCollection` has already cleaned up it's temporary files on close, and the belt-and-suspenders of deleting on exit fails because the files are no longer there.

@tfenne @yfarjoun What Tim said. The SortingCollectionTest reliably succeeds, but then the test runner throws like crazy when shutting down.
```
htsjdk.samtools.util.RuntimeIOException:
  java.nio.file.NoSuchFileException:
    /var/folders/yk/9xqbctq514x_46wjqyh6294jt97pxd/T/tbl/SortingCollectionTest/sortingcollection.8451775603646045897.tmp
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:374)
Caused by: java.nio.file.NoSuchFileException:
  /var/folders/yk/9xqbctq514x_46wjqyh6294jt97pxd/T/tbl/SortingCollectionTest/sortingcollection.8451775603646045897.tmp
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at htsjdk.samtools.util.IOUtil$DeletePathThread.run(IOUtil.java:372)
```

--- a/src/main/java/htsjdk/samtools/util/IOUtil.java
+++ b/src/main/java/htsjdk/samtools/util/IOUtil.java
@@ -360,16 +360,16 @@ public class IOUtil {
      *
      * @see #deleteOnExit(Path)
      */
-    protected static final class DeletePathThread extends Thread {
+    static final class DeletePathThread extends Thread {
 
         private final Path path;
 
-        protected DeletePathThread(Path path) {this.path = path;}
+        DeletePathThread(Path path) {this.path = path;}
 
         @Override
         public void run() {
             try {
-                Files.delete(path);
+                Files.deleteIfExists(path);
             } catch (IOException e) {
                 throw new RuntimeIOException(e);
             }

